tools/bin/commands/util/common_funcs

```diff
@@ -125,69 +135,195 @@ open_url() {
+    git fetch upstream master
+    echo -n "git rebase upstream/master"
+    if ! git rebase upstream/master; then
+        echo " (failed)"
+        echo "git stash"
+        git stash
+        echo "git rebase upstream/master"
+        git rebase upstream/master
+        echo "git stash pop"
+        git stash pop
+    else
+        echo
+    fi
 }
 
 # returns `"true"` if subtree path has changes
 has_changes() {
-  local subtree_path="${1:-}"
-  local ref="${2:-master}"
-  local hash="${CIRCLE_SHA1:-$(git rev-parse HEAD)}"
+    local subtree_path="${1:-}"
+    local ref="${2:-master}"
+    local hash="${CIRCLE_SHA1:-$(git rev-parse HEAD)}"
 
-  local changed_paths="$(git diff ..${ref} --name-only | tr '\n' ' ') $(git status --porcelain=v1 | cut -c 4- | tr '\n' ' ')"
+    local changed_paths="$(git diff ..${ref} --name-only | tr '\n' ' ') $(git status --porcelain=v1 | cut -c 4- | tr '\n' ' ')"
 
-  # when on `master` on CircleCI, compute changed paths for the commit being built
-  if [ "master" == "${CIRCLE_BRANCH:-}" ]; then
-    changed_paths="$changed_paths $(git log $hash^..$hash --name-only --pretty='format:' | tr '\n' ' ')"
-  fi
-  if [[ " $changed_paths" == *" ${subtree_path}"* ]]; then
-    echo "true"
-  fi
+    # when on `master` on CircleCI, compute changed paths for the commit being built
+    if [ "master" == "${CIRCLE_BRANCH:-}" ]; then
+        changed_paths="$changed_paths $(git log $hash^..$hash --name-only --pretty='format:' | tr '\n' ' ')"
+    fi
+    if [[ " $changed_paths" == *" ${subtree_path}"* ]]; then
+        echo "true"
+    fi
+}
+
+#
+# Download `platform-detect` binary
+#
+get_platform_detect_bin() {
+    local version=${1:-$PLATFORM_DETECT_VERSION}
+    local bin_name="platform-detect"
+    local binary="${bin_name}-${version}"
+
+    check_error $(check_for_command "mktemp")
+    check_error $(check_for_command "mkdir")
+    check_error $(check_for_command "curl")
+    check_error $(check_for_command "tar")
+    check_error $(check_for_command "rm")
+    check_error $(check_for_command "mv")
+
+    #
+    # If its already downloaded, skip download
+    #
+    local config_dir=$(syndesis_config_dir)
+    [ -d ${config_dir}/bin ] || mkdir -p ${config_dir}/bin
+
+    local detect_command="${config_dir}/bin/${binary}"
+    if [ -e ${detect_command} ]; then
+        echo ${detect_command}
+        return
+    fi
+
+    #
+    # Check for proper operating system
+    #
+    local os="linux"
+    if $(isMacOs); then
+        os="mac"
+    elif $(isWindows); then
+        os="windows"
+    fi
+
+    #
+    # Download from site
+    #
+    local base_url="https://github.com/syndesisio/syndesis/releases/download/$version"
+    local download_url="${base_url}/platform-detect-$version-$os-amd64.tar.gz"
+    local archive=$(mktemp ${config_dir}/platform-detect-${version}.tar-XXXX)
+    curl -sL --fail -o ${archive} ${download_url} 2>/dev/null || { check_error "ERROR: Could not download ${binary} from ${download_url}"; }
```

Comment:
A release has not yet been made hence there is no `platform-detect` in any releases yet. So to get around I do what you did (although I soft-link rather than copy).

Reply:
Would be interesting to document it, so users should build and soft-link or copy the binary with the version suffix.